### PR TITLE
Feature - Add try catch around the upload_checker call

### PR DESCRIPTION
**Pull request by trgiangdo**
Resolves the comment in https://github.com/Avaiga/taipy/issues/1487?notification_referrer_id=NT_kwDOAj2Ni7QxMTQwOTgzODI3NDozNzU4ODM2Mw#issuecomment-2321588642 by Fred.

If any Exception raised by the `upload_checker`, we log an error and return `InvalidUploadFile` reason

### CODE REVIEW

File: taipy/core/data/_file_datanode_mixin.py [hunk 1]
```diff
@@ -135,7 +135,16 @@ def _upload(self, path: str, upload_checker: Optional[Callable[[str, Any], bool]
             return reason_collection
 
         if upload_checker is not None:
-            if not upload_checker(upload_path.name, upload_data):
+            try:
+                can_upload = upload_checker(upload_path.name, upload_data)
+            except Exception as err:
```
Comment (FredLL-Avaiga) [inline taipy/core/data/_file_datanode_mixin.py:162]: Should the exception be added to the reason?
Especially useful on user function

**Review by FredLL-Avaiga — APPROVED** [review]
Tx

File: taipy/core/data/_file_datanode_mixin.py (hunk 1 above)
Comment (jrobinAV) [reply taipy/core/data/_file_datanode_mixin.py:162]: We could add an optional exception attached to a reason. It is useful information for developers. It is already in the logs, but making it available in taipy so they can use it in the code makes sense to me.

However, by default, I would not display it in the data node viewer to end-users. 

What do you think?

**Review by jrobinAV — APPROVED** [review]

File: taipy/core/data/_file_datanode_mixin.py (hunk 1 above)
Comment (trgiangdo) [reply taipy/core/data/_file_datanode_mixin.py:162]: There is only 1 way of returning a Reason, so I think it's the default.

I'm not sure if we can give the user multiple choices

**Pull request merged**